documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1589,6 +1589,15 @@ if they exceed the configured maximum length. The limit is configurable via the
 `junit.jupiter.params.displayname.argument.maxlength` configuration parameter and defaults
 to 512 characters.
 
+If you'd like to change the default name template for all parameterized tests in your project, you can add the following
+configuration to `junit-platform.properties`
+
+[source,properties,indent=0]
+----
+junit.jupiter.params.displayname.template = {index}
+----
+
+NOTE: `junit.jupiter.params.displayname.template` can override the `name` of `ParameterizedTest` for all test cases.
```

Comment:
Actually, the logic should be inverted: `ParameterizedTest.name()` should override the config parameter. Please also document the config parameter in the Javadoc of `ParameterizedTest.name()`.

Reply:
That's correct: `ParameterizedTest.name()` must override the config parameter in all cases.

But.... to make that possible, we will need to change the `default` value of `ParameterizedTest.name()`.

If we leave it declared as `String name() default DEFAULT_DISPLAY_NAME;` (i.e., the status quo), it will be impossible to know if the user explicitly set the `name` attribute to the value of `DEFAULT_DISPLAY_NAME` intentionally.

Thus, we will need to change the default of the `name` attribute from `DEFAULT_DISPLAY_NAME` to something like `"{default_display_name}"`.

If `"{default_display_name}"` is returned when invoking `name()`, we will:

1. Look up the new config param, and if it has been set, use it.
2. If the new config param has not been set, we will use the value of the `DEFAULT_DISPLAY_NAME` constant for backwards compatibility.

Otherwise, the user explicitly supplied a pattern for the `name` attribute (or their code was compiled against a previous version of the API with `DEFAULT_DISPLAY_NAME` set as the default value for `name`), and we use whatever explicit pattern is set.